### PR TITLE
Add Instances to Interview Status Dashboard

### PR DESCRIPTION
**Pull request by jujulcrane**
### Summary 

This pull request sorts interview statuses into separate instances and adds the ability for users to create brand‐new interview‐status instances. 

* **Feature:**

  * Created `InterviewStatusBase` page that sorts applicants into instances.
  * Added a “+ New Instance” button below the instance list in `InterviewStatusBase`.
  * Implemented `handleNewEmptyInstance()` to prompt the user for a name and create an empty instance with that name.
  * Implemented sorting that ensures that the applicants' statuses are always in a consistent order on the dashboard. 
  * Added a delete instances endpoint in the InterviewStatusAPI

### Test Plan 

https://github.com/user-attachments/assets/86a84c25-65a8-4ce2-ba03-fa61edaf6c8a

1. **Test plan**

   * Load the base screen: verify the “+ New Instance” button renders below existing cards
   * Click “+ New Instance”: a browser prompt should appear asking for the new instance name
   * Enter a unique name: confirm that the UI shows a loading indicator, then a new card with that name appears
   * Click the new card: verify that `InterviewStatusDashboard` opens with an empty statuses list
   * Select an instance and click delete instance button: verify that the instance is deleted and no longer appears as a card

2. **Edge cases**

   * Dismiss prompt or submit a blank name: verify nothing breaks and no new instance is created
   * Submit a duplicate name: verify that error is displayed to user

### Notes
 * I plan to improve the UI and change the window prompt into an actual form 
 * I plan to implement uploading an instance from a CSV and from a candidate decider instance

### CODE REVIEW

**Review by clementroze — COMMENTED** [review]
Hi Juju! 

Thank you for the detailed test plan! 

I tested it on localhost and everything seems to work!
- "+ New Instance" button renders correctly, clicking makes a new instance
- Your two edge cases (duplicate/empty name) work are handled well
- "Back to Instances" button works correctly

One small note: is it intended that creating a new Instance with no applicants inside doesn't save it?

See video below – I create a new Instance called "no applicants here", but then I immediately click "Back to Instances" and it doesn't show up in the dashboard.

https://github.com/user-attachments/assets/e0151615-3cdf-4927-939b-5e3e6d1b1688


Also, I understand that you are planning on changing the browser alert into an actual modal. But i also have a few other notes regarding overall design + usage of components. I'll leave these notes in a second review! :)

**Review by clementroze — COMMENTED** [review]
hi again! Ok so here are some thoughts I had regarding design & accessibility.

None of them are strict requirements for this PR – if you feel like the goals of your PR have been met, please feel free to move ahead first! I just thought it'd be nice to mention them for reference.

However, one thing I do think we should try to fix (maybe not in this PR, but at least eventually) are the cards. If the one from Semantic UI isn't meeting our needs (proper HTML semantics, keyboard navigation, styling etc.), then we should probably make a custom component. 

We would then use it in all cases where we need a card (eg: Forms, Candidate Decider, Interview Status Dashboard, etc.) for consistency.

Let me know if you have any questions or want to discuss anything!

File: frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          you think this is a mistake.
+        </h1>
+      ) : (
+        <Card.Group>
+          {groups.map((group) => (
+            <Card
+              onClick={() => setSelected(group)}
+              key={group.instanceName}
+              className={styles.card}
+            >
```
[the remark this answers is not in the document]
Comment (jujulcrane) [reply frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx:83]: Semantic lets me pass as="button" type="button" as I prop to  the Card component - would that solve these problems? It appears to be focusable now.

File: frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  if (isLoading) {
+    return <div className={styles.loading}>Loading...</div>;
+  }
+
+  if (selected) {
+    return (
+      <div>
+        <Button onClick={() => setSelected(null)} basic>
+          Back to Instances
+        </Button>
```
[the remark this answers is not in the document]
Comment (jujulcrane) [reply frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx:59]: Will do - I moved it to the top right but i'm not sure if its better. Let me know if you have a better idea for where I should move the button.

File: frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.module.css [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+.loading {
+  text-align: center;
+  font-size: 1.5rem;
+  color: #555;
+}
+
+.card {
+  cursor: pointer;
+  border-radius: 0.5rem !important;
+  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1) !important;
```
[the remark this answers is not in the document]
Comment (jujulcrane) [reply frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.module.css:16]: You have a good point, I'll remove the styles to stay consistent.

**Comment by jujulcrane** [comment]
> Hi Juju!
> 
> Thank you for the detailed test plan!
> 
> I tested it on localhost and everything seems to work!
> 
> * "+ New Instance" button renders correctly, clicking makes a new instance
> * Your two edge cases (duplicate/empty name) work are handled well
> * "Back to Instances" button works correctly
> 
> One small note: is it intended that creating a new Instance with no applicants inside doesn't save it?
> 
> See video below – I create a new Instance called "no applicants here", but then I immediately click "Back to Instances" and it doesn't show up in the dashboard.
> 
>  Screen.Recording.2025-05-08.at.12.17.45.AM.mov 
> Also, I understand that you are planning on changing the browser alert into an actual modal. But i also have a few other notes regarding overall design + usage of components. I'll leave these notes in a second review! :)

hm for now I was thinking that it shouldn't save if there are no applicants because I don't see why anyone would create an empty instance? It is a bit complicated to allow empty instances since with the way the database schema is set up, I would need to create a dummy applicant since the instanceName is stored in each specific applicant if that makes sense. So if no applicants have that instanceName, it won't show up as an instance. Although if you think it is important to allow empty instances I'm sure I could find a way to implement it. Also, I plan to implement a way to create a new instance with a csv or by exporting from candidate decider so once that is implemented I predict that will be used more than "create new empty instance"

File: frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.module.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (clementroze) [reply frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.module.css]: OK sounds good! But to reiterate – i'm happy to change the styles if you want to (for example i think the box shadow improvement is nice!). We would just have to do it for all the cards on the other pages too for consistency

Feel free to revisit this at another time tho, not urgent at all :)

File: frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (clementroze) [reply frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx]: Oh perfect, yes i see it now, and i can also use [Tab] to navigate around.

Screenshot 2025-05-12 at 6 30 06 PM


also ur right to use `button`!

File: frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (clementroze) [reply frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx]: new button components look great 🤩 

i think "Back" buttons are typically placed on the left side since ppl read from left to right. So people would naturally look towards the left to go "backwards" or to return to a page, mirroring the way humans read information.

I think the "Delete" button can remain on the right tho?

Something like this could work:

Screenshot 2025-05-12 at 6 34 26 PM

[the remark this answers is not in the document]
Comment (clementroze) [reply frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx]: the buttons should have the same margins as the main content on the page, so maybe move the button container into the `dashboardContainer` div?

**Comment by clementroze** [comment]
> hm for now I was thinking that it shouldn't save if there are no applicants because I don't see why anyone would create an empty instance? It is a bit complicated to allow empty instances since with the way the database schema is set up, I would need to create a dummy applicant since the instanceName is stored in each specific applicant if that makes sense. So if no applicants have that instanceName, it won't show up as an instance. Although if you think it is important to allow empty instances I'm sure I could find a way to implement it. Also, I plan to implement a way to create a new instance with a csv or by exporting from candidate decider so once that is implemented I predict that will be used more than "create new empty instance"



@jujulcrane Ya you're totally right that this is an edge case and that realistically ppl wouldn't create empty instances. Was just thinking about the edge case, but if it's complicated technically to fix this, then no worries. 

i see what u mean about the `instanceName`, tbh the dummy applicant workaround probably seems a bit too much, at least for this initial PR? 

And yeah sounds good for the CSV file, makes sense to me that admins will be using the export feature from Cand Decider

**Review by clementroze — APPROVED** [review]
Nice! LGTM! 

I think if you could just get [this change too](https://github.com/cornell-dti/idol/pull/955/#discussion_r2085629412) that would be sweet, but no worries if not. 

Thank you!! :shipit: :shipit: :shipit: :shipit:

File: frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jujulcrane) [reply frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx]: Oops didnt see this - will do

File: frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className={styles.container}>
+      {!groups || groups.length === 0 ? (
+        <h1>
+          You currently do not have access to any interview status instances! Please contact{' '}
```
Comment (cchrischen) [inline frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx:95]: This is already behind `admin` middleware, so the error message should be something like "No dashboards yet". Users also can't create new instances if there are no existing instances.
image

**Review by cchrischen — COMMENTED** [review]
good work!

File: frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx (hunk 4 above)
Comment (cchrischen) [reply frontend/src/components/Admin/InterviewStatus/InterviewStatusBase.tsx:95]: nvm you can just address this in your new PR
